pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,408 @@
+	// E.g.,
+	//     PodSandboxConfig.LogDirectory = `/var/log/pods/<podUID>/`
+	//     ContainerConfig.LogPath = `containerName_Instance#.log`
+	//
+	// WARNING: Log managment and how kubelet should interface with the
+	// container logs are under active discussion in
+	// https://issues.k8s.io/24677. There *may* be future change of direction
+	// for logging as the discussion carries on.
+	LogDirectory string
+	// Labels are key value pairs that may be used to scope and select
+	// individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map that may be set by external
+	// tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string
+
+	// Linux contains configurations specific to Linux hosts.
+	Linux *LinuxPodSandboxConfig
+}
+
+// Labels are key value pairs that may be used to scope and select individual
+// resources.
+// Label keys are of the form:
+//     label-key ::= prefixed-name | name
+//     prefixed-name ::= prefix '/' name
+//     prefix ::= DNS_SUBDOMAIN
+//     name ::= DNS_LABEL
+type Labels map[string]string
+
+// LinuxPodSandboxConfig holds platform-specific configuraions for Linux
+// host platforms and Linux-based containers.
+type LinuxPodSandboxConfig struct {
+	// CgroupParent is the parent cgroup of the sandbox. The cgroupfs style
+	// syntax will be used, but the container runtime can convert it to systemd
+	// semantices if needed.
+	CgroupParent string
+	// NamespaceOptions contains configurations for the sandbox's namespaces.
+	// This will be used only if the PodSandbox uses namespace for isolation.
+	NamespaceOptions NamespaceOptions
+}
+
+// NamespaceOptions provides options for Linux namespaces.
+type NamespaceOptions struct {
+	// HostNetwork uses the host's network namespace.
+	HostNetwork bool
+	// HostPID uses the host's pid namesapce.
+	HostPID bool
+	// HostIPC uses the host's ipc namespace.
+	HostIPC bool
+}
+
+// DNSOptions specifies the DNS servers and search domains.
+type DNSOptions struct {
+	// Servers is a list of DNS servers of the cluster.
+	Servers []string
+	// Searches is a list of DNS search domains of the cluster.
+	Searches []string
+}
+
+type PodSandboxState string
+
+const (
+	// PodSandboxReady means the sandbox is functioning properly.
+	PodSandboxReady PodSandboxState = "Ready"
+	// PodSandboxInNotReady means the sandbox is not functioning properly.
+	PodSandboxNotReady PodSandboxState = "NotReady"
+)
+
+// PodSandboxFilter is used to filter a list of PodSandboxes.
+type PodSandboxFilter struct {
+	// Name of the sandbox.
+	Name *string
+	// ID of the sandbox.
+	ID *PodSandboxID
+	// State of the sandbox.
+	State *PodSandboxState
+	// LabelSelector to select matches.
+	// Only api.MatchLabels is supported for now and the requirements
+	// are ANDed. MatchExpressions is not supported yet.
+	LabelSelector unversioned.LabelSelector
+}
+
+// PodSandboxListItem contains minimal information about a sandbox.
+type PodSandboxListItem struct {
```

Comment:
It has minimal information as the comment suggests, so it'd be misleading to do that.